src/v/transform/rpc/client.cc

```diff
@@ -317,27 +318,29 @@ client::do_store_wasm_binary_once(
 
 ss::future<result<stored_wasm_binary_metadata, cluster::errc>>
 client::do_local_store_wasm_binary(
-  iobuf data, model::timeout_clock::duration timeout) {
+  model::wasm_binary_iobuf data, model::timeout_clock::duration timeout) {
     return _local_service->local().store_wasm_binary(std::move(data), timeout);
 }
 
 ss::future<result<stored_wasm_binary_metadata, cluster::errc>>
 client::do_remote_store_wasm_binary(
-  model::node_id node, iobuf data, model::timeout_clock::duration timeout) {
-    auto resp = co_await _connections->local()
-                  .with_node_client<impl::transform_rpc_client_protocol>(
-                    _self,
-                    ss::this_shard_id(),
-                    node,
-                    timeout,
-                    [timeout, data = std::move(data)](
-                      impl::transform_rpc_client_protocol proto) mutable {
-                        return proto.store_wasm_binary(
-                          store_wasm_binary_request(std::move(data), timeout),
-                          ::rpc::client_opts(
-                            model::timeout_clock::now() + timeout));
-                    })
-                  .then(&::rpc::get_ctx_data<store_wasm_binary_reply>);
+  model::node_id node,
+  model::wasm_binary_iobuf data,
+  model::timeout_clock::duration timeout) {
+    auto resp
+      = co_await _connections->local()
+          .with_node_client<impl::transform_rpc_client_protocol>(
+            _self,
+            ss::this_shard_id(),
+            node,
+            timeout,
+            [timeout, data = std::move(data)](
+              impl::transform_rpc_client_protocol proto) mutable {
+                return proto.store_wasm_binary(
+                  store_wasm_binary_request(std::move(*data()), timeout),
```

Comment:
Ah moving this data out breaks things. I need to implement custom serde or something for this

Reply:
done